src/Model.js

```diff
@@ -509,8 +513,25 @@ const Model = class Model {
             }
         }
 
-        this._initFields(Object.assign({}, this._fields, mergeObj));
-        this._refreshMany2Many(m2mRelations); // eslint-disable-line no-underscore-dangle
+        const mergedFields = {
+            ...this._fields,
+            ...mergeObj,
+        };
+
+        const updatedModel = new ThisModel(this._fields);
+        updatedModel._initFields(mergedFields); // eslint-disable-line no-underscore-dangle
+
+        // determine if model would have different related models after update
+        updatedModel._refreshMany2Many(m2mRelations); // eslint-disable-line no-underscore-dangle
+        const relationsEqual = Object.keys(m2mRelations).every(name =>
```

Comment:
I'm a bit unsure if this is actually sufficient as regards `relationsEqual` which is calculated using the idea that in `Model#_refreshMany2Many` we only ever actually touch the state when `arrayDiffActions` returns a non-null object.